### ISSUE TITLE
manual/scalaGuide/main/async/ScalaAsync.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#597]
- https://github.com/garbagetown/playdocja/blob/2.2.0/documentation/2.2.0/manual/scalaGuide/main/async/ScalaAsync.md

```
--- /Users/garbagetown/Desktop/2.1.5/manual/scalaGuide/main/async/ScalaAsync.md 2013-09-19 22:53:02.000000000 +0900
+++ /Users/garbagetown/Desktop/2.2.0/manual/scalaGuide/main/async/ScalaAsync.md 2013-09-19 10:11:22.000000000 +0900
@@ -14,56 +14,26 @@

 To create a `Future[Result]` we need another future first: the future that will give us the actual value we need to compute the result:

-'''scala
-val futurePIValue: Future[Double] = computePIAsynchronously()
-val futureResult: Future[Result] = futurePIValue.map { pi =>
-  Ok("PI value computed: " + pi)    
-}
-'''
+@[future-result](code/ScalaAsync.scala)

 All of Play’s asynchronous API calls give you a `Future`. This is the case whether you are calling an external web service using the `play.api.libs.WS` API, or using Akka to schedule asynchronous tasks or to communicate with actors using `play.api.libs.Akka`.

 Here is a simple way to execute a block of code asynchronously and to get a `Future`:

-'''scala
-val futureInt: Future[Int] = scala.concurrent.Future {
-  intensiveComputation()
-}
-'''
-
-> **Note:** Here, the intensive computation will just be run on another thread. It is also possible to run it remotely on a cluster of backend servers using Akka remote.
-
-## AsyncResult
-
-While we were using `SimpleResult` until now, to send an asynchronous result, we need an `AsyncResult` to wrap the actual `SimpleResult`:
-
-'''scala
-def index = Action {
-  val futureInt = scala.concurrent.Future { intensiveComputation() }
-  Async {
-    futureInt.map(i => Ok("Got result: " + i))
-  }
-}
-'''
+@[intensive-computation](code/ScalaAsync.scala)

-> **Note:** `Async { }` is an helper method that builds an `AsyncResult` from a `Future[Result]`.
+It's important to understand which thread code runs on with futures.  In the two code blocks above, there is an import on Plays default execution context.  This is an implicit parameter that gets passed to all methods on the future API that accept callbacks.  The execution context will often be equivalent to a thread pool, though not necessarily.
+
+## Returning futures
+
+While we were using the `Action.apply` builder methods to build actions until now, to send an asynchronous result, we need to use the `Action.async` buider method:
+
+@[async-result](code/ScalaAsync.scala)

 ## Handling time-outs

 It is often useful to handle time-outs properly, to avoid having the web browser block and wait if something goes wrong. You can easily compose a promise with a promise timeout to handle these cases:

-'''scala
-
-def index = Action {
-  val futureInt = scala.concurrent.Future { intensiveComputation() }
-  val timeoutFuture = play.api.libs.concurrent.Promise.timeout("Oops", 2.seconds)
-  Async {
-    Future.firstCompletedOf(Seq(futureInt, timeoutFuture)).map { 
-      case i: Int => Ok("Got result: " + i)
-      case t: String => InternalServerError(t)
-    }  
-  }
-}
-'''
+@[timeout](code/ScalaAsync.scala)

 > **Next:** [[Streaming HTTP responses | ScalaStream]]
\ No newline at end of file

```

**Comment by garbagetown** [comment]
pull reqed at #687

**Comment by wm3** [comment]
merged #687
